konnectd/ui/src/containers/Goodbye/Goodbyescreen.js

```diff
@@ -0,0 +1,125 @@
+    position: 'relative',
+    display: 'inline-block'
+  }
+});
+
+class Goodbyescreen extends React.PureComponent {
+  componentDidMount() {
+    this.props.dispatch(executeHello());
+  }
+
+  render() {
+    const { classes, hello } = this.props;
+
+    const loading = hello === null;
+    return (
+      <ResponsiveScreen loading={loading}>
+        {renderIf(hello !== null && !hello.state)(() => (
+          <div>
+            <Typography variant="h5" component="h3">
+              <FormattedMessage id="konnect.goodbye.headline" defaultMessage="Goodbye"></FormattedMessage>
+            </Typography>
+            <Typography variant="subtitle1" className={classes.subHeader}>
+              <FormattedMessage id="konnect.goodbye.subHeader"
+                defaultMessage="you have been signed out from your Kopano account">
+              </FormattedMessage>
+            </Typography>
+            <Typography gutterBottom>
+              <FormattedMessage id="konnect.goodbye.message.close"
+                defaultMessage="You can close this window now.">
+              </FormattedMessage>
+            </Typography>
+          </div>
+        ))}
+        {renderIf(hello !== null && hello.state === true)(() => (
+          <div>
+            <Typography variant="h5" component="h3">
+              <FormattedMessage
+                id="konnect.goodbye.confirm.headline"
+                defaultMessage="Hello {displayName}"
+                values={{displayName: hello.displayName}}>
+              </FormattedMessage>
+            </Typography>
+            <Typography variant="subtitle1" className={classes.subHeader}>
+              <FormattedMessage id="konnect.goodbye.confirm.subHeader"
+                defaultMessage="please confirm sign out">
+              </FormattedMessage>
+            </Typography>
+
+            <Typography gutterBottom>
+              <FormattedMessage id="konnect.goodbye.message.confirm"
+                defaultMessage="Press the button below, to sign out from your Kopano account now.">
+              </FormattedMessage>
+            </Typography>
+
+            <DialogActions>
+              <div className={classes.wrapper}>
+                <Button
+                  color="secondary"
+                  className={classes.button}
+                  onClick={(event) => this.logoff(event)}
+                >
+                  <FormattedMessage id="konnect.goodbye.signoutButton.label"
+                    defaultMessage="Sign out"></FormattedMessage>
+                </Button>
+              </div>
+            </DialogActions>
+          </div>
+        ))}
+      </ResponsiveScreen>
+    );
+  }
+
+  logoff(event) {
+    event.preventDefault();
+
+    this.props.dispatch(executeLogoff()).then((response) => {
+      const { history } = this.props;
+
+      if (response.success) {
+        this.props.dispatch(executeHello());
+        history.push('/goodbye');
+      }
+    });
+  }
+}
+
+Goodbyescreen.propTypes = {
```

Comment:
Is using TS an option? https://www.typescriptlang.org/docs/handbook/migrating-from-javascript.html#moving-to-typescript-files

Reply:
I'd very much love that! Not sure if I can introduce in one PR not only react but TS as well though 🙈